src/tool/idl/main.cpp

```diff
@@ -467,8 +489,21 @@ struct writer : writer_base<writer>
             write(")]");
         }
     }
+
+    void write(indentation indent_level)
```

Comment:
The python projection writer already has an implementation for automatically injecting indententation after a newline. I think that's a better design than having to explicitly write indentation. Might be worth factoring that out to make it easier for other languag projections to consume.

Reply:
Thanks for clarification. When looking how library works I've initially looked at cppwinrt (because I knew what it is supposed to do) and then at idl, since it was the smallest and easiest to understand. I've seen how python tool was post-processing write buffer to indent multi-line output. I can take this approach for idl. Even if tool is used only as sample code.

Reply:
If you're going to work in IDL project, do you mind also moving it over to samples dir?